src/Process/LayeredMaterialNodeProcessing.js

```diff
@@ -0,0 +1,309 @@
+        ancestor,
+        redraw: !force,
+    };
+
+    return context.scheduler.execute(command).then(
+        (result) => {
+            if (node.material === null) {
+                return;
+            }
+            if (!result) {
+                // shouldn't happen
+                // eslint-disable-next-line no-console
+                console.warn(`Invalid value received. Probably a bug in the provider (${layer.protocol})`);
+                return;
+            }
+
+            const level = ancestor ? ancestor.level : node.level;
+            // Assign .level to texture
+            if (Array.isArray(result)) {
+                for (let j = 0; j < result.length; j++) {
+                    result[j].texture.level = level;
+                }
+
+                node.setTexturesLayer(result, l_COLOR, layer.id);
+            } else if (result.texture) {
+                result.texture.level = level;
+                node.setTexturesLayer([result], l_COLOR, layer.id);
+            }
+
+            node.layerUpdateState[layer.id].success();
+
+            return result;
+        },
+        (err) => {
+            if (err instanceof CancelledCommandException) {
+                node.layerUpdateState[layer.id].success();
+            } else {
+                node.layerUpdateState[layer.id].failure(Date.now());
+                // TODO event
+                context.scene.notifyChange(node.layerUpdateState[layer.id].secondsUntilNextTry() * 1000, false);
+            }
+        });
+}
+
+export function updateLayeredMaterialNodeElevation(context, layer, node) {
+    _updateLayeredMaterialNodeElevation(context, layer, node);
+}
+
+function _updateLayeredMaterialNodeElevation(context, layer, node, parent) {
+    // TODO: we need either
+    //  - compound or exclusive layers
+    //  - support for multiple elevation layers
+
+    // Elevation is currently handled differently from color layers.
+    // This is caused by a LayeredMaterial limitation: only 1 elevation texture
+    // can be used (where a tile can have N textures x M layers)
+    const ts = Date.now();
+    let ancestor = null;
+
+    const currentElevation = node.materials[RendererConstant.FINAL].getElevationLayerLevel();
+
+    // Step 0: currentElevevation is -1 BUT material.loadedTexturesCount[l_ELEVATION] is > 0
+    // means that we already tried and failed to download an elevation texture
+    if (currentElevation == -1 && node.materials[0].loadedTexturesCount[l_ELEVATION] > 0) {
+        return;
+    }
+
+    // First step: if currentElevation is empty (level is -1), we *must* use the texture from
+    // one of our parent. This allows for smooth transitions when subdividing
+    // We don't care about layer status (isLayerFrozen) or limits (tileInsideLimit) because
+    // we simply want to use ancestor's texture with a different pitch
+    if (currentElevation == -1) {
+        ancestor = findAncestorWithValidTextureForLayer(node, parent || node.parent, l_ELEVATION);
```

Comment:
As said above, I'd prefer not to modify to much the code, because I'll have to rebase over PR #244